src/utils/havoc.js

```diff
@@ -446,8 +446,34 @@ class ObjectValueHavocingVisitor {
   }
 
   visitAbstractValue(val: AbstractValue): void {
-    for (let i = 0, n = val.args.length; i < n; i++) {
-      this.visitValue(val.args[i]);
+    if (!val.mightBeObject()) {
+      // Only objects need to be havoced.
+      return;
+    }
+    if (val.values.isTop()) {
+      // If we don't know which object instances it might be,
+      // then it might be one of the arguments that created
+      // this value. See #2179.
```

Comment:
Here I get concerned. By definition ALL objects known to prepack might be denoted by val. Just havocing the arguments does not seem sufficient. As things are today, we refuse to compile any program that accesses a property of an abstract object with a values domain that is Top.